test/extended/builds/start.go

```diff
@@ -127,17 +131,20 @@ var _ = g.Describe("[builds][Slow] starting a build using CLI", func() {
 
 		g.It("should accept --from-repo with --commit as input", func() {
 			g.By("starting the build with a Git repository")
-			// NOTE: This actually takes the commit from the origin repository. If the
-			// test-build-app changes, this commit has to be bumped.
-			br, err := exutil.StartBuildAndWait(oc, "sample-build", "--commit=f0f3834", fmt.Sprintf("--from-repo=%s", exampleBuild))
+			gitCmd := exec.Command("git", "rev-parse", "HEAD")
```

Comment:
do HEAD~1 so we're not just building HEAD.  otherwise the git commit matching check later is suspect.